libdnf/rpm/reldep.cpp

```diff
@@ -85,7 +85,7 @@ ReldepId Reldep::get_reldep_id(
 }
 
 ReldepId Reldep::get_reldep_id(const BaseWeakPtr & base, const std::string & reldep_str, int create) {
-    if (reldep_str[0] == '(') {
+    if (is_rich_dependency(reldep_str)) {
         // Rich dependency
```

Comment:
I would remove this line now, because the function name `is_rich_dependency` is quite self-explaining.